lib/compress/zstd_ldm.c

```diff
@@ -576,6 +569,15 @@ size_t ZSTD_ldm_blockCompress(rawSeqStore_t* rawSeqStore,
     /* Input positions */
     BYTE const* ip = istart;
 
+    /* If using opt parser, use LDMs only as candidates rather than always accepting them */
+    if (cParams->strategy >= ZSTD_btopt) {
+        size_t lastLLSize;
+        ms->ldmSeqStore = *rawSeqStore;
```

Comment:
Why not use a pointer for the `ldmSeqStore`, since we're effectively modifying the original `rawSeqStore`?

Reply:
So apparently `*rawSeqStore` is can be something allocated on the stack or heap (depending on multithreading in `buildSeqStore()`). So if `ms->ldmSeqStore` were a pointer, we'd lose track of the `*rawSeqStore` it points to once we call `blockCompressor()` - which is why we just copy the `*rawSeqStore`.

Reply:
> So if ms->ldmSeqStore were a pointer, we'd lose track of the *rawSeqStore it points to once we call blockCompressor()

What do you mean by that?

Reply:
In `ZSTD_buildSeqStore()`, there are two ways to trigger the ldm codepath: 1) there exists a `zc->externSeqStore` (ldm is enabled with multithreading), or 2) ldm is enabled, and is allocated on the stack as `rawSeqStore_t ldmSeqStore = {NULL, 0, 0, 0, 0};`.

So in the second case, if `ms->ldmSeqStore` is a pointer to the `ldmSeqStore` object, then once we call `blockCompressor(ms, seqStore, rep, src, srcSize);`, because the pointer to the object itself isn't in the function arguments, it seems that by entering the new scope of `blockCompressor()`, the pointer doesn't seem to point to anything that's valid anymore. Or at least that's what I think is happening. A long time ago I did have `ms->ldmSeqStore` as a pointer and ran into this bug where the `ldmSeqStore` did not persist into the call to `blockCompressor()`.